sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClientIntegrationTest.java

```diff
@@ -326,4 +369,256 @@ void autoRenewLockOnReceiveMessage() {
             receiver.close();
         }
     }
+
+    @Test
+    void testBasicReceiveAndDeleteWithBinaryData() {
+        // Arrange
+        final String messageTrackingId = UUID.randomUUID().toString();
+        final ServiceBusMessage message = TestUtils.getServiceBusMessageBinary(messageTrackingId, 0, 64);
+
+        // Assert & Act
+        StepVerifier.create(sender.send(message).thenMany(receiveDeleteModeReceiver.receive()))
+            .assertNext(receivedMessage ->
+                Assertions.assertTrue(receivedMessage.getProperties().containsKey(MESSAGE_TRACKING_ID)))
+            .expectNoEvent(Duration.ofSeconds(2))
+            .thenCancel()
+            .verify();
+    }
+
+    @Test
+    void testBasicReceiveAndCompleteWithLargeBinaryData() {
```

Comment:
This is testing the service rather than our APIs. I think we should be conscious of which service tests we bring over.

Reply:
I am not convinced that adding "Basic" brings any value.
`receiveAndComplete()`
`receiveAndCompleteBySequenceNumber()`
is sufficient to describe the scenario.